### ISSUE TITLE
Creating custom train, test, eval dataset

### ISSUE THREAD

**Issue by aidiss** [zou-group/textgrad#114]
Trying to reuse `Tutorial-Prompt-Optimization.ipynb` example for my case.

In the example notebook there is this line

```py
train_set, val_set, test_set, eval_fn = load_task("BBH_object_counting", evaluation_api=llm_api_eval)
```

How to build my own train, val, test dataset; and eval fn.
What are the datatypes used and how to construct them manually?

**Comment by mertyg** [comment]
Hey @aidiss ! You can see some examples [here](https://github.com/zou-group/textgrad/blob/main/textgrad/tasks/big_bench_hard.py#L79). The training/test set are basically a list of question / ground truth answer pairs.

**Comment by JiaYuanChng** [comment]
Hello,

First off - amazing work and paper! I am looking to apply this to my own work, and would appreciate some guidance on setting up the dataset structure.

In the [prompt optimization tutorial](https://github.com/zou-group/textgrad/blob/bf5b0c5df0d029293dcc355a389a55908c61a9c6/examples/notebooks/Tutorial-Prompt-Optimization.ipynb), the `train_set` is an object of `BigBenchHard`, which is later used as input in `tg.tasks.DataLoader(train_set, batch_size=3, shuffle=True)`.

I wanted to clarify if the procedure of setting up our own train, val and test sets from `our_own_dataset.csv` file is as follows:
E.g. After splitting into a separate `train_our_own_dataset.csv`:
```
x,y
"Some text1", 9
"Some text2", 2
"Some text3", 4
```
Create a new class:
```
import pandas as pd
import textgrad as tg
from textgrad.tasks.base import DataSet

class MyOwnData(DataSet):
    def __init__(self, csvfile): 
        self.data = pd.read_csv(csvfile) 

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index):
        row = self.data.iloc[index]
        return row["x"], row["y"]

train_set = MyOwnData(train_our_own_dataset.csv)
train_loader = tg.tasks.DataLoader(train_set, batch_size=3, shuffle=True)
```
